src/meta/src/rpc/elections.rs

```diff
@@ -0,0 +1,838 @@
+        Ok(v) => v,
+        _ => return None,
+    };
+    Some((current_leader_info, current_leader_lease))
+}
+
+/// Retrieve infos about the current leader
+/// Wrapper for `get_infos`
+///
+/// ## Returns
+/// None on error, else infos about the leader and lease
+async fn get_infos_obj<S: MetaStore>(
+    meta_store: &Arc<S>,
+) -> Option<(MetaLeaderInfo, MetaLeaseInfo)> {
+    match get_infos(meta_store).await {
+        None => None,
+        Some(infos) => {
+            let (leader, lease) = infos;
+            return Some((
+                MetaLeaderInfo::decode(&mut leader.as_slice()).unwrap(),
+                MetaLeaseInfo::decode(&mut lease.as_slice()).unwrap(),
+            ));
+        }
+    }
+}
+
+fn gen_rand_lease_id(addr: String) -> u64 {
+    let mut ds = DefaultHasher::new();
+    addr.hash(&mut ds);
+    ds.finish()
+    // FIXME: We are unable to use a random lease at the moment
+    // During testing, meta gets killed, new meta starts
+    // meta detects that lease is still there, with same addr, but diff ID
+    // meta believes that leader is out there and becomes follower
+    // IMHO we can only use random lease id, if we have at least 2 meta nodes
+    // https://github.com/risingwavelabs/risingwave/issues/6844
+    // rand::thread_rng().gen_range(0..std::u64::MAX)
+}
+
+/// Used to manage single leader setup. `run_elections` will continuously run elections to determine
+/// which nodes are **leaders** and which are **followers**.
+///
+/// To become a leader a **follower** node **campaigns**. A follower only ever campaigns if it
+/// detects that the current leader is down. The follower becomes a leader by acquiring a lease
+/// from the **meta store**. After getting elected the new node will start its **term** as a leader.
+/// A term lasts until the current leader crashes.   
+///
+/// ## Arguments
+/// `addr`: Address of the current leader, e.g. "127.0.0.1".
+/// `meta_store`: Holds information about the leader.
+/// `lease_time_sec`: Time in seconds that a lease will be valid for.
+/// A large value reduces the meta store traffic. A small value reduces the downtime during failover
+///
+/// ## Returns:
+/// `MetaLeaderInfo` containing the leader who got elected initially.
+/// `JoinHandle` running all future elections concurrently.
+/// `Sender` for signaling a shutdown.
+/// `Receiver` receiving true if this node got elected as leader and false if it is a follower.
+pub async fn run_elections<S: MetaStore>(
+    addr: String,
+    meta_store: Arc<S>,
+    lease_time_sec: u64,
+) -> MetaResult<(
+    MetaLeaderInfo,
+    JoinHandle<()>,
+    Sender<()>,
+    Receiver<(HostAddr, bool)>,
+)> {
+    // Randomize interval to reduce mitigate likelihood of simultaneous requests
+    let mut rng: StdRng = SeedableRng::from_entropy();
+    let mut ticker = tokio::time::interval(
+        Duration::from_secs(lease_time_sec / 2) + Duration::from_millis(rng.gen_range(0..500)),
+    );
```

Comment:
Unfortunately the ticker will extends the meta start time in deterministic recovery test. This will result in the need to extend the sleep interval between test queries, it will take longer for the whole deterministic recovery test. This is the reason for the current test failure.
Actually as it said before, the randomize interval for sending campaign request is really not necessary. Because even though when etcd receives multi-requests in the same time, its internal locks will also ensure that this time one and only one will get elected. 
Anyway, so the quickest way to fix it is to remove the ticker for `initial_election` (remove `reset` also works, because the ticker for `initial_election` will complete immediately as before). The ticker is still needed for `election` and `term` loop.